File: CodingSamples/Java/Platform/FileChannelTest/Transformer.java
class Transformer {

	public static void transform(java.nio.ByteBuffer bytes) {
		for(int i = 0, j = bytes.capacity() - 1; i < j; ++i, --j) {
			byte ib = bytes.get(i);
			byte jb = bytes.get(j);
			bytes.put(i, jb);
			bytes.put(j, ib);
		}
	}
}

File: CodingSamples/DotNet/Platform/MemMapFileTest/DemoApp/Transformer.cs
class Transformer
{
    public static void Transform(UnmanagedMemoryAccessor data)
    {
        for(long i = 0, j = data.Capacity - 1; i < j; ++i, --j)
        {
            byte ib = data.ReadByte(i);
            byte jb = data.ReadByte(j);
            data.Write(i, jb);
            data.Write(j, ib);
        }
    }
}